### PR TITLE
build: upgrade golang to 1.9.1 and fix build issue

### PR DESCRIPTION
**Pull request by bassam**
Addresses #909

### CODE REVIEW

File: build/makelib/golang.mk [hunk 1]
```diff
@@ -118,36 +118,16 @@ endif
...
-endef
-$(foreach p,$(GO_TEST_PACKAGES),$(eval $(call go.test.project,$(lastword $(subst /, ,$(p))),$(p),CGO_ENABLED=0,$(GO_STATIC_FLAGS))))
-$(foreach p,$(GO_LONGHAUL_TEST_PACKAGES),$(eval $(call go.test.project,$(lastword $(subst /, ,$(p))),$(p),CGO_ENABLED=0,$(GO_STATIC_FLAGS))))
-
-
 .PHONY: go.build
-go.build: go.build.packages go.build.test.packages
+go.build:
+	@echo === go build $(PLATFORM)
+	$(foreach p,$(GO_STATIC_PACKAGES),@CGO_ENABLED=0 $(GO) build -v -i -o $(GO_OUT_DIR)/$(lastword $(subst /, ,$(p)))$(GO_OUT_EXT) $(GO_STATIC_FLAGS) $(p)${\n})
```
Comment (jbw976) [inline build/makelib/golang.mk]: what functional purpose does the addition of the `\n` newline do?

File: build/makelib/golang.mk [hunk 2]
```diff
@@ -118,36 +118,16 @@ endif
...
-go.build.packages: go.build.packages.$(1)
-
-go.install.packages.$(1):
-	@echo === go install $(1) $(PLATFORM)
-	@$(3) $(GO) install -v $(4) $(2)
-go.install.packages: go.install.packages.$(1)
-endef
-$(foreach p,$(GO_STATIC_PACKAGES),$(eval $(call go.project,$(lastword $(subst /, ,$(p))),$(p),CGO_ENABLED=0,$(GO_STATIC_FLAGS))))
-
-define go.test.project
```
Comment (travisn) [inline build/makelib/golang.mk]: we don't use `go.test.project` anymore?

File: build/makelib/golang.mk (hunk 1 above)
Comment (bassam) [reply build/makelib/golang.mk]: a recipe line must end in a newline, and I want generate the newline here.

File: build/makelib/golang.mk (hunk 2 above)
Comment (bassam) [reply build/makelib/golang.mk]: it was an intermediate define to run test project builds in parallel. we now run them sequentially so not needed.

**Pull request merged**